File: examples/java/kora-java-resilient/src/main/java/io/koraframework/example/resilient/FallbackService.java
package io.koraframework.example.resilient;

import io.koraframework.common.annotation.Component;
import io.koraframework.resilient.fallback.annotation.Fallback;

@Component
public class FallbackService {

    public static final String VALUE = "OK";
    public static final String FALLBACK = "FALLBACK";

    @Fallback(method = "getFallback()")
    public String getValue(boolean fail) {
        if (fail)
            throw new IllegalStateException("Failed");

        return VALUE;
    }

    protected String getFallback() {
        return FALLBACK;
    }
}

File: examples/kotlin/kora-kotlin-resilient/src/main/kotlin/io/koraframework/kotlin/example/resilient/FallbackService.kt
package io.koraframework.kotlin.example.resilient

import io.koraframework.common.annotation.Component
import io.koraframework.resilient.fallback.annotation.Fallback
import java.util.concurrent.ThreadLocalRandom

@Component
open class FallbackService {
    @Fallback(method = "getFallback()")
    open fun getValue(fail: Boolean): String {
        if (fail) {
            throw IllegalStateException("Failed")
        }
        return VALUE
    }

    protected open fun getFallback(): String = FALLBACK

    companion object {
        const val VALUE = "OK"
        const val FALLBACK = "FALLBACK"
    }
}
